inc/packages/admin/namespace.php

```diff
@@ -527,3 +634,34 @@ function maybe_add_data_to_description( $description, $plugin ) {
 	$description .= '</p><p class="authors"><em>' . sprintf( __( 'Hosted on %1$s', 'fair' ), esc_html( $repo_host ) ) . '</em>';
 	return $description;
 }
+
+/**
+ * Filters the theme description when preparing themes for JS.
+ *
+ * @param array $themes Array of themes prepared for JS.
+ * @return array Array of themes with possible modifications.
+ */
+function maybe_add_data_to_theme_description( $themes ) {
+	foreach ( $themes as &$theme ) {
+		$did = get_file_data( get_stylesheet_directory() . '/style.css', [ 'ThemeID' => 'Theme ID' ] )['ThemeID'];
+		if ( empty( $did ) || ! str_starts_with( $did, 'did:plc:' ) ) {
+			continue;
+		}
+
+		$repo_host = Info\get_repository_hostname( $did );
+		if ( empty( $repo_host ) ) {
+			continue;
+		}
+
+		/* translators: %1$s: repository hostname */
+		$additional_description = '<p class="authors"><em>' . sprintf( __( 'Hosted on %1$s', 'fair' ), esc_html( $repo_host ) ) . '</em>';
```

Comment:
It's not incorrect, but do we need the specificity as there is only a single text replacement? We could just use `%s`.